policies/compatibility.rego

```diff
@@ -0,0 +1,352 @@
+package comparison_after_resolution
+
+import rego.v1
+
+# Semantic Convention Registry Compatibility Checker
+#
+# This file contains rules for checking backward compatibility
+# between different versions of semantic convention registries.
+# It builds upon the data structures and rules defined in the
+# semconv package.
+
+# Import the previous release (baseline) and current sets of attributes, metrics.
+baseline_attributes := [attr |
+    some g in data.semconv.registry_baseline_groups
+    some attr in g.attributes
+]
+registry_attributes := [attr |
+    some g in data.semconv.registry_groups
+    some attr in g.attributes
+]
+registry_attribute_names := {attr.name |
+    some g in data.semconv.registry_groups
+    some attr in g.attributes
+}
+baseline_metrics := [ g |
+    some g in data.semconv.baseline_groups
+    g.type == "metric"
+]
+registry_metrics := [ g |
+    some g in data.semconv.groups
+    g.type == "metric"
+]
+registry_metric_names := { g.metric_name | some g in registry_metrics }
+
+
+# Rules we enforce:
```

Comment:
would you mind adding some tests? (thanks @MadVikingGod for making it possible!)

Reply:
Having lots of trouble making that happen actually.  The new tests don't look like what was set up before_resolution, and the errors are pretty hard to deal with.

I'll take some more time to try to figure this out, but perhaps someone can help me better understand error output *or* we need to bake this feature into weaver...

Reply:
Alright, I was able to get things working, but it's a bit wonky.  You have to understand how weaver helper methods work to actually get tests working.